test/sharness/t0080-repo.sh

```diff
@@ -0,0 +1,86 @@
+#!/bin/sh
+#
+# Copyright (c) 2014 Jeromy Johnson
+# MIT Licensed; see the LICENSE file in this repository.
+#
+
+test_description="Test ipfs repo operations"
+
+. lib/test-lib.sh
+
+test_init_ipfs
+test_launch_ipfs_daemon
+
+test_expect_success "'ipfs add afile' succeeds" '
+	echo "some text" > afile
+	HASH=`ipfs add -q afile`
+	echo -n $HASH > hashfile
+'
+
+test_expect_success "added file was pinned" '
+	ipfs pin ls -type=recursive | grep `cat hashfile`
+'
```

Comment:
- dont need `hashfile`, use `$HASH`
- save the output and display it if not there, so we can debug:
```
	ipfs pin ls -type=recursive | grep "$HASH" > actual ||
	fsh cat actual
```

- or maybe want to make sure the output matches _exactly_:

```
	echo "$HASH recursive" > pin_ls_expected &&
	ipfs pin ls -type=recursive > pin_ls_actual &&
	test_cmp pin_ls_expected pin_ls_actual
```
- yeah, and actually, we want to check the output of all combinations is exactly what we want:

```sh
ipfs pin ls 
ipfs pin ls --type=recursive
ipfs pin ls --type=direct
ipfs pin ls --type=indirect
ipfs pin ls --type=recursive -q
ipfs pin ls --type=direct -q 
ipfs pin ls --type=indirect -q
```